api/controllers/recipes.controller.js

```diff
@@ -26,6 +24,30 @@ exports.getAllRecipes = async(req, res, next) => {
     }
 }
 
+exports.getSingleRecipeFromUrl = async(req, res) => {
+
+    if(!req.params.id) {
+        return res.status(400).json({status: 400, message: "Missing recipe ID"})
+    }
+
+    const recipeUrl = req.params.id;
+
+    try {
+        const recipe = await RecipeService.getSingleRecipeFromUrl(recipeUrl);
+        return res.status(200).json({
+            status: 200,
+            data: recipe,
+            message: "Recipe Found"
+        });
+    } catch(e) {
+        return res.status(400).json({
+            status: 400,
```

Comment:
`status: 400` as an attribute is redundant, `.status(400)` takes care of that. Please do the same with any other places like this.